clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1105,6 +1105,19 @@ static ParmVarDecl *getSyclKernelHandlerArg(FunctionDecl *KernelCallerFunc) {
   return (KHArg != KernelCallerFunc->param_end()) ? *KHArg : nullptr;
 }
 
+static bool isReadOnlyAccessor(const TemplateArgument &AccessModeArg) {
+  const auto *AccessModeArgEnumType =
+      AccessModeArg.getIntegralType()->castAs<EnumType>();
+  EnumDecl *ED = AccessModeArgEnumType->getDecl();
+
+  auto IsReadOnly = llvm::find_if(ED->enumerators(), [&](EnumConstantDecl *E) {
+    return E->getName() == "read" &&
+           E->getInitVal() == AccessModeArg.getAsIntegral();
+  });
+
+  return (IsReadOnly != ED->enumerator_end()) ? true : false;
```

Comment:
Er... you don't need a ternary to change 1 boolean to another.  
```suggestion
  return IsReadOnly != ED->enumerator_end();
```

Reply:
Hahaha! I have no idea why I did this. Thanks for catching it